incubator/elasticsearch/templates/client-deployment.yaml

```diff
@@ -21,7 +21,8 @@ spec:
 {{ toYaml .Values.client.podAnnotations | indent 8 }}
         {{- end }}
     spec:
-      serviceAccountName: {{ template "elasticsearch.fullname" . }}
+      securityContext:
```

Comment:
As of ES 6.0 Elasticsearch requires the config directory and data directory to be owned by the ES user, this ensures the mounted volumes are owned by ES. :)

Reply:
@mikn In some implementations of k8s (namely SUSE CaaSP) you need to explicitly grant priviledged mode to the service-account via RBAC, otherwise master won't start.